src/python/arcor2_arserver/rpc/robot.py

```diff
@@ -190,23 +197,26 @@ async def register(
 @scene_needed
 async def register_for_robot_event_cb(req: srpc.r.RegisterForRobotEvent.Request, ui: WsClient) -> None:
 
-    ensure_scene_started()
+    async with ctx_write_lock(req.args.robot_id, "ui.owner"):
```

Comment:
Neměl by tady být read zámek? Klient se zde registruje k odebírání pózy/jointů robota. Nic se tedy nemění.

Reply:
Ono se tam sahá do globálních proměnných, tím write zámkem se třeba ošetří přístup k `ROBOT_JOINTS_REGISTERED_UIS`, ale je pravda, že to se bude měnit, tak tam hned dám read

Reply:
Vidíte, to jsem si neuvědomil... Jo, to `ROBOT_JOINTS_REGISTERED_UIS` by určitě bylo dobré nějak předělat. Ale kdy a jestli se k tomu dostanu... :-)